### PR TITLE
Update doc comment tests that contain XML comments with paths

### LINKED ISSUE #8807: Failing Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests
Affected PRs
- https://github.com/dotnet/roslyn/pull/8802
- https://github.com/dotnet/roslyn/pull/8796
- https://github.com/dotnet/roslyn/pull/8812
- https://github.com/dotnet/roslyn/pull/8801
- https://github.com/dotnet/roslyn/pull/8799
- https://github.com/dotnet/roslyn/pull/8646
- https://github.com/dotnet/roslyn/pull/8435

Test Result (12 failures / +12)
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationModeTests.XmlSyntaxError_Included
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.IncludedName_DuplicateNameAttribute
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.PragmaDisableWarningInXmlFile

- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.WRN_XMLParseIncludeError_Source
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XDocument_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XmlError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Bug684184
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_IOError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_FileNotFound_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathNotFound_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_Cycle_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathError_WRN_XMLDocBadFormedXML

### PR DESCRIPTION
**Pull request by cston**
`XComments` with "--" in the comment are serialized as "- -" since "--" is not valid within a XML comment. This occurs when writing a doc comment containing an error from ```<include file='[path.xml]'/>``` when ```[path.xml]``` contains "--".

This change updates the tests to match the comment serialization. The alternative, to change how errors are reported from the doc comment compiler, is a more significant change.

Fixes #8807

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler, @amcasey

File: src/Test/Utilities/Desktop/TestHelpers.cs [hunk 1]
```diff
@@ -123,5 +125,25 @@ public class TestAnalyzer : DiagnosticAnalyzer
 
             return dir.CreateFile(assemblyName + ".dll").WriteAllBytes(analyzerCompilation.EmitToArray());
         }
+
+        /// <summary>
+        /// <see cref="System.Xml.Linq.XComment.Value"/> is serialized with "--" replaced by "- -"
+        /// </summary>
+        public static string AsXmlCommentText(string text)
+        {
+            var builder = new StringBuilder();
```
Comment (amcasey) [inline src/Test/Utilities/Desktop/TestHelpers.cs]: This seems over-optimized for a test helper.  Maybe use `string.Replace`?

If keeping this version, consider using a pooled builder.

File: src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs [hunk 2]
```diff
@@ -70,7 +69,7 @@ private static List<HandlerSpan> GetHandlerSpans(ImmutableArray<Cci.ExceptionHan
             {
                 HandlerSpan span;
 
-                if (region.HandlerKind == ExceptionRegionKind.Filter)
+                if (region.HandlerKind == System.Reflection.Metadata.ExceptionRegionKind.Filter)
```
Comment (amcasey) [inline src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs]: Why?

**Comment by amcasey** [comment]
LGTM.  As we discussed offline, the key thing is that the path is unmangled in the diagnostic.

File: src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs (hunk 2 above)
Comment (cston) [inline src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs]: `ILOpCode` (used elsewhere in this file) is defined in two namespaces, one of which is `System.Reflection.Metadata.ILOpCode`, so I removed `using System.Reflection.Metadata;`.

Comment (tmat) [inline src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs]: You can add 
```using System.Reflection.Metadata;``` inside namespace definition in the file to resolve that ambiguity. That's the general pattern used elsewhere until we move to SRM v1.3

Comment (AlekseyTs) [inline src/Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs]: It looks like the only changes in this file is the removal of using and a fix-ups caused by that. If that is the case. Please revert the change, at least exclude it from this PR as it looks unrelated to the issue being addressed.

**Comment by AlekseyTs** [comment]
LGTM, modulo seemingly unrelated changes in ...Test/Utilities/Shared/Metadata/ILBuilderVisualizer.cs

File: src/Test/Utilities/Desktop/TestHelpers.cs (hunk 1 above)
Comment (cston) [inline src/Test/Utilities/Desktop/TestHelpers.cs]: Using `string.Replace` would probably require multiple calls to handle `"---"`. The current implementation is more verbose but maybe clearer. Will leave as is.
